http-client/src/main/java/io/airlift/http/client/jetty/JettyHttpClient.java

```diff
@@ -942,8 +942,7 @@ public <T, E extends Exception> HttpResponseFuture<T> executeAsync(Request reque
 
         JettyResponseFuture<T, E> future = new JettyResponseFuture<>(request, jettyRequest, requestSize::getBytes, responseHandler, span, stats, recordRequestComplete);
 
-        BufferingResponseListener listener = new BufferingResponseListener(future, Ints.saturatedCast(request.getMaxContentLength().orElse(maxContentLength).toBytes()))
-        {
+        JettyResponseListener<T, E> listener = new JettyResponseListener<>(jettyRequest, future, Ints.saturatedCast(request.getMaxContentLength().orElse(maxContentLength).toBytes())) {
             @Override
             public void onBegin(Response response)
```

Comment:
It's weird that we have our own subclass, yet create another subclass of it here. We should either move this method into `JettyResponseListener` or inline it here.